Library Rename, Simplified Setup, and Improved Documentation

RoachxD opened the pull request:
This pull request introduces several improvements and changes to the `steam-community-market` library, including renaming the library, simplifying the setup process, updating the example file, and enhancing documentation. The main changes are as follows:

1. Library Renaming and Version Bump:

    - Renamed the library from `steam_community_market` to `steam-community-market`.
    - Updated the library description.
    - Bumped the library version to `1.3.0`.

2. Currencies and Enums Update:

    - Moved currencies from `enums.py` to `currencies.py`.
    - Renamed and better documented currency-related classes.
    - Improved documentation for `AppID` enum.

3. Exceptions Enhancement:

    - Renamed and better documented `LegacyCurrencyException`.
    - Implemented a new exception called `InvalidCurrencyException`.

4. Market Class Update:

    - Adapted the class for `Python 3.9`.
    - Improved documentation with valid Sphinx directives.
    - Refactored and renamed various methods.
    - Enhanced the logic for currency exception handling.

5. Simplified Setup Process:

    - Simplified `setup.py` by including metadata and other options inside `setup.cfg`.
    - Added a Python version check to raise a `RuntimeError` if the version is less than `3.9`.
    - Populated `setup.cfg` with more project-relevant information.

6. Updated `example.py`:

    - Made compatibility updates for version `1.3.0`.

These changes aim to improve the overall experience of using the `steam-community-market` library, making it more user-friendly, and easier to maintain.

RoachxD committed:
`1.3.0` finish!
  - `__init__.py`:
    - Renamed library from `steam_community_market` to `steam-community-market`.
    - Changed the description of the library to something more fitting.
    - Changed the version of the library to `1.3.0`.
    - Added `Currencies`, `Enums`, `Exceptions` and `Market` to the `__all__` variable.
    - Removed `Requests` from the `__all__` variable.
    - Removed all other attributes as they will be added inside `setup.cfg`.
  - `currencies.py`:
    - Moved currencies from `enums.py`.
    - Renamed `ESteamSupportedCurrency` to `SteamCurrency` and populated it with the legacy currencies as well.
    - Renamed `ESteamUnsupportedCurrency` to `SteamLegacyCurrency`.
    - Better documented both of them.
  - `enums.py`:
    - Better documented `AppID` enum.
  - `exceptions.py`:
    - Renamed `SteamUnsupportedCurrency` to `LegacyCurrencyException` and better documented it.
    - Implemented a new exception called `InvalidCurrencyException`.
  - `market.py`:
    - Adapted the class for `Python 3.9`.
    - Better documented the entire class using valid Sphinx directives.
    - Removed `has_invalid_name` and instead fixed the name directly inside `get_overview`.
    - Modified the accepted format for `get_overviews_from_dict`.
    - Renamed `fix_name` to `_fix_item_name` and made it static.
    - Renamed `price_to_float` to `_price_to_float` and made it static.
    - Improved the logic of `_supported_currency` for detecting and raising both `InvalidCurrencyException` and `LegacyCurrencyException`.

RoachxD committed:
Simplified setup!
  - Simplified `setup.py` by including the `metadata` and other `options` inside `setup.cfg`.
  - Added a Python version check which will raise a `RuntimeError` if the version value is less than `3.9`.
  - Populated the `setup.cfg` with more information relevant to the project.

RoachxD committed:
Updated `example.py`!
  - Compatibility update for `1.3.0`.

Gobot1234 commented:
I think renaming the library is a rather daft, you install the library using `pip install steam_community_market`, you import it using `steam_community_market`, why should it be renamed?

Gobot1234 commented on steam_community_market/exceptions.py:
```diff
@@ -1,25 +1,74 @@
+        :type currency: Any
+        """
+
+        self.currency = currency
+
+    def __str__(self) -> str:
+        """Return a string representation of the exception.
+
+        This method returns a string that indicates which currency is not supported
+        by the Steam Community Market.
+
+        :return: A string representation of the exception.
+        :rtype: str
+        """
+
+        return f'Currency "{self.currency}" is not supported by the Steam Community Market.'
```
Typically you would just use `super().__init__(f'Currency "{self.currency}" is not supported by the Steam Community Market.')` and I wouldn't personally bother this this method and docstring

RoachxD replied:
From my perspective it would be nice to have everything documented. @offish, what do you say?

Gobot1234 commented on steam_community_market/exceptions.py:
```diff
@@ -1,25 +1,74 @@
 
         self.currency = currency
 
     def __str__(self) -> str:
         """Return a string representation of the exception.
 
+        This method returns a string that indicates which legacy currency is not supported
+        by the Steam Community Market anymore.
+
```
Again, I wouldn't personally bother documenting this

RoachxD replied:
We'll have to, once gain, agree to disagree. @offish, what do you say?

Gobot1234 commented on steam_community_market/market.py:
```diff
@@ -1,84 +1,66 @@
-    def get_overview(self, name: str, app_id: int | AppID) -> dict | None:
-        """Gets the prices and volume of an item.
+        .. versionadded:: 1.0.0
+        .. versionchanged:: 1.3.0
+            * Moved :param:`app_id`'s validation to :meth:`_app_id_value`.
+            * Improved type hinting for both parameters and return value.
+            * Removed :param:`name`'s validation check and just called :meth:`_fix_item_name`.
+            * Simplified the return statement.
```
Typically since the end user doesn't care about changes like this they shouldn't be included. If you want to have a history of these kinds of changes use better git commit titles

RoachxD replied:
I thought it would be nice to have a changelog for the latest `.. versionchanged::` directive.
For example, if the above method would change in `1.3.2`, the changelog related to `1.3.0` will get removed and replaced by the changelog of `1.3.2`.

Gobot1234 commented on steam_community_market/market.py:
```diff
@@ -1,84 +1,66 @@
+        :type name: str
         :param app_id: The AppID of the game the item is from.
-        :type app_id: :class:`int`, :class:`AppID`
+        :type app_id: int or AppID
```
```suggestion
        :type app_id: :class:`int` | :class:`AppID`
```
Use the standard syntax for this, I'm also not sure why you have removed the class role from these because that will potentially break inter-sphinx and make the documentation worse

RoachxD replied:
According to this, the syntax should be fine: https://www.sphinx-doc.org/en/master/tutorial/describing-code.html#cross-referencing-python-objects

Also, yes.. The `:class:` directive should be re-added for classes defined inside this library, my bad.

Gobot1234 commented on steam_community_market/market.py:
```diff
@@ -274,60 +245,117 @@ def get_volume(self, name: str, app_id: int | AppID) -> int | None:
+        if isinstance(app_id, AppID):
+            return app_id.value
+        if isinstance(app_id, int) or (
+            support_lists and isinstance(app_id, list[Union[int, AppID]])
```
isinstance checks don't work with GenericAliases, this will fail at runtime. Runtime typechecking like this is a lost cause

RoachxD replied:
Indeed, I'll be fixing this shortly.

Gobot1234 commented on steam_community_market/market.py:
```diff
@@ -274,60 +245,117 @@ def get_volume(self, name: str, app_id: int | AppID) -> int | None:
+        :rtype: SteamCurrency
+        """
+
+        result: Union[SteamCurrency, None] = None
+        if isinstance(currency, int) and currency in SteamCurrency.__members__.values():
+            result = SteamCurrency(currency)
+
+        elif (
+            isinstance(currency, str)
+            and (currency := currency.upper()) in SteamCurrency.__members__.keys()
+        ):
+            result = SteamCurrency[currency]
+
+        elif isinstance(currency, SteamCurrency):
+            result = currency
+
+        if result is None:
+            raise InvalidCurrencyException(currency)
+
+        elif (
+            isinstance(currency, SteamLegacyCurrency)
+            or result.name in SteamLegacyCurrency.__members__.keys()
+            or result.value in SteamLegacyCurrency.__members__.values()
+        ):
+            raise LegacyCurrencyException(currency)
+
+        return result
```
```suggestion
        if isinstance(currency, int) and currency in SteamCurrency.__members__.values():
            return SteamCurrency(currency)

        elif (
            isinstance(currency, str)
            and (currency := currency.upper()) in SteamCurrency.__members__.keys()
        ):
            return SteamCurrency[currency]

        elif isinstance(currency, SteamCurrency):
            return currency

        raise LegacyCurrencyException(currency)

```

RoachxD replied:
You have miss-interpreted the two exceptions and when they should get raised:
- InvalidCurrencyException gets raised when the user inputs a non-existent currency, that couldn't be supported by Steam.
- LegacyCurrencyException gets raised when the user inputs one of the legacy currencies, that are no longer supported by Steam:
  - 33 - SEK / Swedish Krona
  - 36 - BYN / Belarusian Ruble
  - 42 - BGN / Bulgarian Lev
  - 43 - HRK / Croatian Kuna
  - 44 - CZK / Czech Koruna
  - 45 - DKK / Danish Krone
  - 46 - HUF / Hungarian Forint
  - 47 - RON / Romanian Leu

Gobot1234 replied:
Oh sorry you're right but, what's the point even accepting legacy currencies if they just raise?

RoachxD replied:
Mainly.. Because I aspire for this library to get as competent as possible in interpreting everything related to the Steam Community Market, so it seems only fair to notify the user that the currency he's trying to use is no longer supported by the platform via an exception.

RoachxD commented:
> I think renaming the library is a rather daft, you install the library using `pip install steam_community_market`, you import it using `steam_community_market`, why should it be renamed?

Because of inconsistency as it actually gets installed via `pip install steam-community-market`. The import remains the same in both cases (e.g. `steam_community_market`).

Gobot1234 commented:
> > I think renaming the library is a rather daft, you install the library using `pip install steam_community_market`, you import it using `steam_community_market`, why should it be renamed?
> 
> Because of inconsistency as it actually gets installed via `pip install steam-community-market`. The import remains the same in both cases (e.g. `steam_community_market`).
```sh
$ pip install --upgrade steam_community_market

Collecting steam_community_market
  Downloading steam_community_market-1.2.3.tar.gz (5.7 kB)
  Preparing metadata (setup.py) ... done
Requirement already satisfied: requests in /opt/homebrew/lib/python3.10/site-packages (from steam_community_market) (2.28.2)
Requirement already satisfied: certifi>=2017.4.17 in /opt/homebrew/lib/python3.10/site-packages (from requests->steam_community_market) (2022.6.15.1)
Requirement already satisfied: charset-normalizer<4,>=2 in /opt/homebrew/lib/python3.10/site-packages (from requests->steam_community_market) (3.0.1)
Requirement already satisfied: idna<4,>=2.5 in /opt/homebrew/lib/python3.10/site-packages (from requests->steam_community_market) (3.4)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in /opt/homebrew/lib/python3.10/site-packages (from requests->steam_community_market) (1.26.14)
Building wheels for collected packages: steam_community_market
  Building wheel for steam_community_market (setup.py) ... done
  Created wheel for steam_community_market: filename=steam_community_market-1.2.3-py3-none-any.whl size=6308 sha256=1e072cd7543c4de5caf6d1484db2058484aaff037e9cdf537a9bbdf76f687175
  Stored in directory: /Users/James/Library/Caches/pip/wheels/93/33/d9/aff1248a0cdaea26b80b04b9fb1bedf8ba0d2a13bfb184c1c6
Successfully built steam_community_market
Installing collected packages: steam_community_market
Successfully installed steam_community_market-1.2.3
```
it doesn't seem that way to me

RoachxD commented:
> > > I think renaming the library is a rather daft, you install the library using `pip install steam_community_market`, you import it using `steam_community_market`, why should it be renamed?
> > 
> > 
> > Because of inconsistency as it actually gets installed via `pip install steam-community-market`. The import remains the same in both cases (e.g. `steam_community_market`).
> 
> ```shell
> $ pip install --upgrade steam_community_market
> 
> Collecting steam_community_market
>   Downloading steam_community_market-1.2.3.tar.gz (5.7 kB)
>   Preparing metadata (setup.py) ... done
> Requirement already satisfied: requests in /opt/homebrew/lib/python3.10/site-packages (from steam_community_market) (2.28.2)
> Requirement already satisfied: certifi>=2017.4.17 in /opt/homebrew/lib/python3.10/site-packages (from requests->steam_community_market) (2022.6.15.1)
> Requirement already satisfied: charset-normalizer<4,>=2 in /opt/homebrew/lib/python3.10/site-packages (from requests->steam_community_market) (3.0.1)
> Requirement already satisfied: idna<4,>=2.5 in /opt/homebrew/lib/python3.10/site-packages (from requests->steam_community_market) (3.4)
> Requirement already satisfied: urllib3<1.27,>=1.21.1 in /opt/homebrew/lib/python3.10/site-packages (from requests->steam_community_market) (1.26.14)
> Building wheels for collected packages: steam_community_market
>   Building wheel for steam_community_market (setup.py) ... done
>   Created wheel for steam_community_market: filename=steam_community_market-1.2.3-py3-none-any.whl size=6308 sha256=1e072cd7543c4de5caf6d1484db2058484aaff037e9cdf537a9bbdf76f687175
>   Stored in directory: /Users/James/Library/Caches/pip/wheels/93/33/d9/aff1248a0cdaea26b80b04b9fb1bedf8ba0d2a13bfb184c1c6
> Successfully built steam_community_market
> Installing collected packages: steam_community_market
> Successfully installed steam_community_market-1.2.3
> ```
> 
> it doesn't seem that way to me

Have you even tried `pip install steam-community-market`? Out of pure curiosity, of course.

Gobot1234 commented:
It installs the same thing, it I just really don't see the point having 2 different names for the same library if it can be avoided

RoachxD commented:
> It installs the same thing, it I just really don't see the point having 2 different names for the same library if it can be avoided

In my opinion, we don't. The name of the library is `steam-community-market` and Python imposes snake-case, so we are bound to import it as `steam_community_market`.

RoachxD committed:
`Market`: Method signatures changes and doc fixes!
- `app_id` parameter has been moved on the first place for methods that had it.
- Fixed a problem regarding the documentation, many thanks to @Gobot1234.
- Fixed `TypeError` exception related to the `list` type inside `_app_id_value`.

RoachxD committed:
Better `examples.py`!
- Wrote more examples in order to include all methods.

RoachxD committed:
Re-wrote `README.rst` under `README.md`!
- Removed `README.rst`.
- Created a more detailed a visually pleasant version under `README.md`.
- Modified `setup.cfg` in order to adapt it to the new changes.

RoachxD committed:
Updated `README.md`!
- Replaced `import steam_community_market` with `from steam_community_market import *`.
- Removed some of the examples and added a link to the `examples module`.
